### ISSUE TITLE
WriteMatrix saves every matrix as SCIRun-Matrix (OpenSUSE)

### ISSUE THREAD

**Issue by johaenns** [SCIInstitute/SCIRun#1440]
When trying to save a matrix in any other format than SCIRun-binary, this choice is ignored. Occurs as soon as the "filter"-dropdown has white background. This usually happens when editing storage location and type using the "save" menu the second time after SCIRun startup.

![saving_bug](https://cloud.githubusercontent.com/assets/18470885/15438492/a6313fc2-1e89-11e6-8a5b-64cd7c67bb28.png)

**Comment by dcwhite** [comment]
From the description, the use case sounds strange. Does the file type work most of the time otherwise? What about in existing saved networks? I verified each different file type is working on Windows; I won't have time to debug this on OpenSUSE today.

**Comment by johaenns** [comment]
Practically, it always looks like in the above picture on opensuse and thus it never works. Since I am not able to set the filetype correctly, it also doesn't work in existing saved networks.

**Comment by dcwhite** [comment]
Try using the python API to modify the filetype. Open the Python console, and enter:
`
scirun_set_module_state('WriteMatrix:0', 'FileTypeName', 'Matlab Matrix')
`
where the first parameter is the module id, and the third parameter is what type you want. This ought to override the UI value once you execute.

**Comment by dcwhite** [comment]
I tried to replicate this in my OpenSUSE VM, and I couldn't. The WriteMatrix file dialog behaved fine, saving in every possible format. The only quirk was it did not change the file extension to match, but the file type was correct each time. @johaenns Is this still an issue?

**Comment by dcwhite** [comment]
@jessdtate Beta or post-beta?

**Comment by jessdtate** [comment]
we haven't been able to replicate it yet.  We can have one of the interns try it.

**Comment by dcwhite** [comment]
Maybe Linux-pro @RubioJr9 can try

**Comment by RubioJr9** [comment]
I can replicate this on Arch. I think this is a more general issue with file saving on linux. If you take a screenshot through ViewScene on linux, it doesn't automatically append .png and the image never saves. You have to manually type .png in the filename. Same goes for .mat in this case with WriteMatrix. This doesn't occur with saving networks but I think it might be because there is only one option.

**Comment by RubioJr9** [comment]
My mistake, I read the description wrong. I thought it was about file extensions like Dan mentioned. I just tried replicating the bug on KDE Plasma, which is the desktop environment OpenSUSE ran back in 2016 and I can't replicate this. I'm guessing this is a bug in the file explorer because SCIRun uses the dialog that's native to the operating system. Either the dialog was fixed in an update(looks very different now) or OpenSUSE uses a different one.

**Comment by dcwhite** [comment]
So as far as saving files on Linux, do you think it's satisfactory and so we can close this one? Thanks for examining.

**Comment by RubioJr9** [comment]
Yes, it's fine to close unless somebody can replicate it on current software. If somebody can, we can make an option so they can use Qt's non-native file dialog. That would at the very least let them side step their desktop environment's file explorer.

I looked into the file extension issue for Linux and it was reported 10 years ago. They fixed it for the non-native dialog in qt4 but it's broken again in qt5. I found 2 methods on their issue tracker that we could use. https://bugreports.qt.io/browse/QTBUG-11352
Say the user saves a matrix called 'A'. Currently it asks if we want to overwrite a current file named 'A', not 'A.mat'

Method 1. We append the file extension after getSaveFileName() returns the name of the file, so we can save it as 'A.mat'. If the user types in 'A', then the dialog would not be able to confirm they are overwriting file 'A.mat', but instead confirm overwriting 'A'. What we could do is check for the file after appending and show a confirmation box. The problem with this is that if 'A.mat' is typed in the dialog with the extension, then there would be 2 pop ups to confirm overwriting the file. However, QFileDialog has a flag to disable confirming overwrites.

Method 2. Don't use getSaveFileName(). We can use QFileDialog directly to set variables, execute the dialog, and return the selected name. We would still have to make our own confirmation box.

While this isn't as critical for WriteMatrix since it saves as a file correctly even without an extension, saving a screenshot in ViewScene without .png doesn't save at all and gives no warning to the user.

**Comment by dcwhite** [comment]
**Triage 2026-06:** `WriteMatrix` saves every matrix as SCIRun-Matrix on OpenSUSE — the Qt native file dialog on some desktop environments doesn't properly propagate the selected file filter back to the application code. The fix is to use Qt's non-native file dialog as a fallback when the native dialog misbehaves. The last comment describes this approach. Low effort; add a `QFileDialog::DontUseNativeDialog` option on Linux.
